src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -461,15 +463,33 @@ async def resample(self, *, one_shot: bool = False) -> None:
                 timeseries from the resampler before calling this method
                 again).
         """
-        while True:
-            await self._wait_for_next_resampling_period()
+        async for _ in Timer.periodic(
+            timedelta(seconds=self._config.resampling_period.total_seconds())
+        ):
+            now = datetime.now(tz=timezone.utc)
+            timer_error = now - self._window_end
+            # We use a tolerance of 10% of the resampling period
+            tolerance = timedelta(
+                seconds=self._config.resampling_period.total_seconds() / 10.0
+            )
+            if timer_error > tolerance:
```

Comment:
I think all this code can go away now and we can just use the drift that the timer returns.

```suggestion
        # We use a tolerance of 10% of the resampling period
        tolerance = timedelta(
            seconds=self._config.resampling_period.total_seconds() / 10.0
        )
        async for drift in Timer.periodic(
            timedelta(seconds=self._config.resampling_period.total_seconds())
        ):
            if drift > tolerance:

Reply:
now and time_error also need to be updated in the logger message, unfortunately I saw this after accepting the change